bootstrap.cmd

```diff
@@ -1,16 +1,66 @@
-:: Quick and dirty bootstrap. 
+@echo off
 
-where /q dotnet.exe || echo Can't find dotnet.exe! Please add to PATH && goto :EOF
-where /q git.exe || echo Can't find git.exe! Please add to PATH && goto :EOF
-set root=%~dp0
-pushd %root%
+REM Bootstrap the jitutils tools:
+REM 1. If this script is run not from within the jitutils directory (e.g., you've downloaded
+REM    a copy of this file directly), then "git clone" the jitutils project first. Otherwise,
+REM    if we can tell we're being run from within an existing "git clone", don't do that.
+REM 2. Build the jitutils tools.
+REM 3. Download (if necessary) clang-format.exe and clang-tidy.exe (used by the jit-format tool).
+
+where /q dotnet.exe
+if %errorlevel% NEQ 0 echo Can't find dotnet.exe! Please install this ^(e.g., from https://www.microsoft.com/net/core^) and add dotnet.exe to PATH && goto :EOF
+
+set __root=%~dp0
+setlocal
+
+REM Are we already in the dotnet/jitutils repo? Or do we need to clone it?
+REM We look for sample_asmdiff.json and sample_format.json files in the current directory
+REM (which is the directory this script was invoked from).
+
+if not exist %__root%sample_asmdiff.json goto clone_jitutils
```

Comment:
I feel like it would be better to run something like git remote -v and grep the output for jitutils.git, rather than check for the existence of a particular file. What if either of these files get removed from the repo?